components/nearby-diagram.js

```diff
@@ -261,20 +271,28 @@ export default function NearbyDiagram({
       };
     });
     const rectX = (start - offsetXForVariant) * baseWidth - textLength / 2 - 2;
+    const rectWidth = textLength + 4;
+    let indexY = 0;
     let rectY = highestLabelY;
-    if (preRectX2 >= rectX) {
-      rectY = preRectY + labelHeight + blankHeight;
+    if (preLabelInfo) {
+      const pre = preLabelInfo.find((item) => item < rectX);
+      if (pre) {
+        indexY = preLabelInfo.indexOf(pre);
+      } else {
+        indexY = preLabelInfo.length;
+      }
```

Comment:
Conditional expression would shorten this a bit, though functionally the same.

Reply:
Will shorten it. Thanks!